apps/server/src/common.module.ts

```diff
@@ -0,0 +1,84 @@
+import { DynamicModule, Global, Module, NotFoundException } from '@nestjs/common';
+import { RabbitMQModule } from '@golevelup/nestjs-rabbitmq';
+import { MikroOrmModule, MikroOrmModuleSyncOptions } from '@mikro-orm/nestjs';
+import { DB_PASSWORD, DB_URL, DB_USERNAME } from '@src/config';
+import { ALL_ENTITIES } from '@shared/domain';
+import { Configuration } from '@hpi-schul-cloud/commons';
+import { Dictionary, IPrimaryKey } from '@mikro-orm/core';
+import { MongoMemoryDatabaseModule } from '@shared/infra/database';
+import { MongoDatabaseModuleOptions } from '@shared/infra/database/mongo-memory-database/types';
+
+export const defaultMikroOrmOptions: MikroOrmModuleSyncOptions = {
+	findOneOrFailHandler: (entityName: string, where: Dictionary | IPrimaryKey) => {
+		// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+		return new NotFoundException(`The requested ${entityName}: ${where} has not been found.`);
+	},
+};
+
+/**
+ * https://www.npmjs.com/package/@golevelup/nestjs-rabbitmq#usage
+ * we want to have the RabbitMQModule globally available, since it provides via a factory the AMQPConnection.
+ * You shall not explicitly declare the AMQPConnection in your modules since it will create a new AMQPConnection which will not be initialized!
+ *
+ * Therefore, the combination of @Global() and export: [RabbitMQModule] is required.
+ */
+@Global()
```

Comment:
That makes the providers (rabbitmq and mikro-orm) injectable in all modules without importing the respective module! "When you want to provide a set of providers which should be available everywhere out-of-the-box..."
https://docs.nestjs.com/modules#global-modules
Do we really want (need) to provide rabbitmq **and** mikroorm to all our services?

Reply:
I tried it doesn't work, we can try again later